### PR TITLE
feat(transactions): app_start_type reading/writing V2

### PR DESCRIPTION
**Pull request by philipphofmann**
Adds reading and writing to app_start_type . For more context, see https://github.com/getsentry/sentry/issues/36927. This PR is based on a previous PR https://github.com/getsentry/snuba/pull/3127, which was reverted with https://github.com/getsentry/snuba/pull/3182 because it caused an incident.

The error was `'NoneType' object has no attribute 'pop'`, https://sentry.io/organizations/sentry/issues/3613141524/?project=300688&query=is%3Aunresolved

Screen Shot 2022-10-03 at 09 55 52

This problem is fixed now by only calling `app_ctx.pop("start_type", None)` if the `app_ctx` is not `None`.

### CODE REVIEW

File: snuba/datasets/transactions_processor.py [hunk 1]
```diff
@@ -385,6 +392,12 @@ def _sanitize_contexts(
         transaction_ctx.pop("hash", None)
         transaction_ctx.pop("exclusive_time", None)
 
+        # The app_start_type is promoted as a column on a query level, no need to store it again
+        # in the context array.
+        app_ctx = sanitized_context.get("app", {})
+        if app_ctx is not None:
```
Comment (philipphofmann) [inline snuba/datasets/transactions_processor.py:398]: This is the actual fix, most of the other code was already reviewed and approved in https://github.com/getsentry/snuba/pull/3127

File: snuba/datasets/entities/transactions.py [hunk 2]
```diff
@@ -79,6 +79,7 @@
         ColumnToColumn(None, "message", None, "transaction_name"),
         ColumnToColumn(None, "title", None, "transaction_name"),
         ColumnToColumn(None, "spans.exclusive_time", None, "spans.exclusive_time_32"),
+        ColumnToColumn(None, "app_start_type", None, "app_start_type"),
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply snuba/datasets/entities/transactions.py:82]: Yes, https://github.com/getsentry/snuba/pull/3124. Already merged.

File: snuba/datasets/transactions_processor.py [hunk 3]
```diff
@@ -248,6 +248,12 @@ def _process_contexts_and_user(
             if context in contexts:
                 del contexts[context]
 
+        appContext = contexts.get("app")
```
Comment (evanh) [inline snuba/datasets/transactions_processor.py:251]: Is there a reason these variables are camel cased?

Comment (philipphofmann) [reply snuba/datasets/transactions_processor.py:251]: Ups, sorry no. I'm used to camel casing in Swift, ObjC, Java, ... I'm going to change that.

File: snuba/datasets/entities/transactions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (philipphofmann) [reply snuba/datasets/entities/transactions.py:82]: Not sure, I removed it.

**Review by evanh — APPROVED** [review]
OK looks good.

**Comment by philipphofmann** [comment]
> OK looks good.

Thank you, @evanh. I think again I should only merge this between 9 am and 5 pm PDT?

**Comment by evanh** [comment]
> I think again I should only merge this between 9 am and 5 pm PDT?

That would be best, thanks.

**Pull request merged**